src/threading/parallel.mjs

```diff
@@ -0,0 +1,159 @@
+'use strict';
+
+/**
+ * Calculates optimal chunk distribution for parallel processing.
+ * Balances work evenly across workers while respecting max chunk size.
+ *
+ * @param {number} itemCount - Total number of items
+ * @param {number} threads - Number of available worker threads
+ * @param {number} maxChunkSize - Maximum items per chunk
+ * @returns {{ chunkSize: number, numChunks: number }}
+ */
+function calculateChunkStrategy(itemCount, threads, maxChunkSize) {
+  // Determine how many chunks we want (ideally one per thread, but not more than items)
+  const targetChunks = Math.min(threads, itemCount);
+
+  // Calculate base chunk size to distribute work evenly
+  const baseChunkSize = Math.ceil(itemCount / targetChunks);
+
+  // Respect the max chunk size limit
+  const chunkSize = Math.min(baseChunkSize, maxChunkSize);
+
+  // Calculate actual number of chunks needed
+  const numChunks = Math.ceil(itemCount / chunkSize);
+
+  return { chunkSize, numChunks };
+}
+
+/**
+ * Splits indices into chunks of specified size.
+ * @param {number} count - Number of items
+ * @param {number} chunkSize - Items per chunk
+ * @returns {number[][]} Array of index arrays
+ */
+function createIndexChunks(count, chunkSize) {
+  const chunks = [];
+
+  for (let i = 0; i < count; i += chunkSize) {
+    const end = Math.min(i + chunkSize, count);
+    chunks.push(Array.from({ length: end - i }, (_, j) => i + j));
+  }
+
+  return chunks;
+}
+
+/**
+ * Creates a ParallelWorker that uses real Node.js Worker threads
+ * for parallel processing of items.
+ *
+ * @param {string} generatorName - Name of the generator (for chunk processing)
+ * @param {import('./index.mjs').default} pool - WorkerPool instance for spawning workers
+ * @param {object} options - Generator options
+ * @returns {ParallelWorker}
+ */
+export default function createParallelWorker(generatorName, pool, options) {
+  const { threads = 1, chunkSize: maxChunkSize = 20 } = options;
```

Comment:
This shouldn't need a default value, since it'll always be defined (the defaults are defined in the CLI configuration)